webview/src/plots/components/comparisonTable/cell/ComparisonTableMultiCell.tsx

```diff
@@ -1,16 +1,23 @@
-import React, { useCallback, useState } from 'react'
-import { useDispatch } from 'react-redux'
+import React, { ChangeEvent, useCallback, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
 import { ComparisonPlot } from 'dvc/src/plots/webview/contract'
 import { ComparisonTableCell } from './ComparisonTableCell'
 import styles from '../styles.module.scss'
 import { changeDisabledDragIds } from '../comparisonTableSlice'
+import { setComparisonMultiPlotValue } from '../../../util/messages'
+import { PlotsState } from '../../../store'
 
 export const ComparisonTableMultiCell: React.FC<{
   path: string
   plot: ComparisonPlot
 }> = ({ path, plot }) => {
-  const [currentStep, setCurrentStep] = useState<number>(0)
+  const values = useSelector(
+    (state: PlotsState) => state.comparison.multiPlotValues
+  )
```

Comment:
Another slightly more responsive but slightly more complex solution is keeping the slider/image step state being handled by React `useState` and only using `multiPlotValues` on mount. Example of how that looks:

```
const values = useSelector(
    (state: PlotsState) => state.comparison.multiPlotValues
  )
const [step, setCurrentStep] = useState(values?.[path]?.[plot.id] || 0)

  useEffect(() => {
    window.clearTimeout(changeDebounceTimer.current)
    changeDebounceTimer.current = window.setTimeout(() => {
      setComparisonMultiPlotValue(path, plot.id, currentStep)
    }, 500)
  }, [path, plot.id, currentStep])

return (
   <div data-testid="multi-image-cell" className={styles.multiImageWrapper}>
      <ComparisonTableCell
        path={path}
        plot={{ id: plot.id, imgs: [plot.imgs[currentStep]] }}
        imgAlt={`${currentStep} of ${path} (${plot.id})`}
      />
      <div
        className={styles.multiImageSlider}
        onMouseEnter={addDisabled}
        onMouseLeave={removeDisabled}
      >
        <label htmlFor={`${plot.id}-step`}>Step</label>
        <input
          name={`${plot.id}-step`}
          min="0"
          max={plot.imgs.length - 1}
          value={currentStep}
          type="range"
          onChange={event => {
            setCurrentStep(Number(event.target.value))
          }}
        />
        <p>{currentStep}</p>
      </div>
    </div>
)
```

Reply:
On further testing, we're going to need to apply this solution to ensure everything stays stable during experiment runs.